pkg/kv/kvserver/replica_rangefeed.go

```diff
@@ -906,6 +961,16 @@ func (r *Replica) handleClosedTimestampUpdateRaftMuLocked(
 				defer func() { <-m.RangeFeedSlowClosedTimestampNudgeSem }()
 				if err := r.ensureClosedTimestampStarted(ctx); err != nil {
 					log.Infof(ctx, `RangeFeed failed to nudge: %s`, err)
+				} else if shouldCancelTooFarBehind {
```

Comment:
Consider: Compare the leaseholder closed TS against ours, attach the closed TS to the lease info response.